### PR TITLE
Add goerli, bsc, optimism, polygon, avalance, fantom, arbitrum network to query balances using BalanceChecker

### LINKED ISSUE #13845: Querying balance from custom networks are slower than mainnet
### Describe the bug

BalanceChecker contract https://github.com/wbobeirne/eth-balance-checker is not applied for popular networks like Goerli, BSC, Optimism, Polygon, Avalanche, Fantom, and Arbitrum which already have networkId hardcoded in Metamask.

While I have deployed the contract for some networks,

https://github.com/wbobeirne/eth-balance-checker/pull/39

The configuration for those networks should be added on Metamask as well.

### Steps to reproduce

1. Add custom network for following networks

- Goerli
- BSC
- Optimism
- Polygon
- Avalanche
- Fantom
- Arbitrum

2. Metamask will send more than 20 balance check requests to those RPC endpoints which cause heavy loads.

### Related issues

https://github.com/MetaMask/metamask-extension/pull/6181
https://github.com/MetaMask/metamask-extension/pull/6049
https://github.com/MetaMask/metamask-extension/issues/6061

### PR to fix the issue

https://github.com/MetaMask/metamask-extension/pull/13846

### PR DESCRIPTION
**Pull request by ghost**
Fixes #13845

Related PR

https://github.com/MetaMask/metamask-extension/pull/6181
https://github.com/MetaMask/metamask-extension/pull/6049
https://github.com/MetaMask/metamask-extension/issues/6061

Added following network's BalanceChecker contract to Metamask

- Goerli
- BSC
- Optimism
- Polygon
- Avalanche
- Fantom
- Arbitrum

### CODE REVIEW

**Comment by 0xAyanami** [comment]
I have read the CLA Document and I hereby sign the CLA

File: app/scripts/constants/contracts.js [hunk 1]
```diff
@@ -6,3 +6,17 @@ export const SINGLE_CALL_BALANCES_ADDRESS_ROPSTEN =
   '0xb8e671734ce5c8d7dfbbea5574fa4cf39f7a54a4';
 export const SINGLE_CALL_BALANCES_ADDRESS_KOVAN =
   '0xb1d3fbb2f83aecd196f474c16ca5d9cffa0d0ffc';
+export const SINGLE_CALL_BALANCES_ADDRESS_GOERLI =
+  '0xb1d3fbb2f83aecd196f474c16ca5d9cffa0d0ffc';
```
Comment (mcmire) [inline app/scripts/constants/contracts.js:10]: This address doesn't match the contract address you listed on https://github.com/wbobeirne/eth-balance-checker/pull/39, is that intentional?

File: app/scripts/constants/contracts.js [hunk 2]
```diff
@@ -6,3 +6,17 @@ export const SINGLE_CALL_BALANCES_ADDRESS_ROPSTEN =
...
+export const SINGLE_CALL_BALANCES_ADDRESS_OPTIMISM =
+  '0xB1c568e9C3E6bdaf755A60c7418C269eb11524FC';
+export const SINGLE_CALL_BALANCES_ADDRESS_POLYGON =
+  '0x2352c63A83f9Fd126af8676146721Fa00924d7e4';
+export const SINGLE_CALL_BALANCES_ADDRESS_AVALANCHE =
+  '0xD023D153a0DFa485130ECFdE2FAA7e612EF94818';
+export const SINGLE_CALL_BALANCES_ADDRESS_FANTOM =
+  '0x07f697424ABe762bB808c109860c04eA488ff92B';
+export const SINGLE_CALL_BALANCES_ADDRESS_ARBITRUM =
+  '0x07f697424ABe762bB808c109860c04eA488ff92B';
```
Comment (mcmire) [inline app/scripts/constants/contracts.js:22]: Same for this.

**Review by mcmire — COMMENTED** [review]
Thanks for this PR. I've left some comments below!

**Comment by 0xAyanami** [comment]
@mcmire Hello, thank you for pointing out, looks like it was a copy-paste error while copying addresses. Now I have double checked those addresses and looks like it is now fixed as it should be.

**Review by mcmire — DISMISSED** [review]
I think this shows that we need to do some refactoring around the account tracking code, but that's no problem. This looks good to me!

**Comment by 0xAyanami** [comment]
@NiranjanaBinoy No problem, here are the links for you to verify those contracts

https://goerli.etherscan.io/address/0x9788C4E93f9002a7ad8e72633b11E8d1ecd51f9b#code

https://bscscan.com/address/0x2352c63A83f9Fd126af8676146721Fa00924d7e4#code

https://optimistic.etherscan.io/address/0xB1c568e9C3E6bdaf755A60c7418C269eb11524FC#code

https://polygonscan.com/address/0x2352c63A83f9Fd126af8676146721Fa00924d7e4#code

https://snowtrace.io/address/0xD023D153a0DFa485130ECFdE2FAA7e612EF94818#code

https://ftmscan.com/address/0x07f697424ABe762bB808c109860c04eA488ff92B#code

https://arbiscan.io/address/0x151E24A486D7258dd7C33Fb67E4bB01919B7B32c#code

**Comment by 0xAyanami** [comment]
@NiranjanaBinoy @mcmire Hello, so any chance to merge this PR yet? Thanks.

**Review by NiranjanaBinoy — DISMISSED** [review]
@ayanamidev Thank you for your contributions and for being patient with our process, I have verified the contracts, and everything looks good to me.

**Review by NiranjanaBinoy — APPROVED** [review]

**Review by mcmire — APPROVED** [review]

**Pull request merged**